zio-http/src/main/scala/zio/http/ZClient.scala

```diff
@@ -630,7 +637,13 @@ object ZClient {
       body: Body,
       sslConfig: Option[ClientSSLConfig],
     )(implicit trace: Trace): ZIO[Scope, Throwable, Response] = {
-      val request = Request(version, method, url, headers, body, None)
+      val requestHeaders = body.mediaType match {
```

Comment:
Would this be the best place for this logic? I am assuming that if you have a `mediaType` attached to your `Body` and no header set with the content type, it should automatically pull the value from the `Body`

Reply:
Okay so from my understanding we should override the header if the body has a media type